### DISCUSSION TITLE
WebGui + Jack standalone on RPI4

### DISCUSSION THREAD

**Discussion by protectorateindustrial** [DISTRHO/Cardinal#473] [Ideas]
I'm not sure if this is a thing already but I didnt find any documentation relating to it, but would it be possible to run Cardinal headless on RPI4 with the GUI as a webserver? Kinda like mod devices and modep do. 

I tested Cardinal on rpi last night and while the audio ran pretty ok over jack, the gui was unusable (1fps, unresponsive), at least over a VNC connection.

**Comment by dromer** [comment]
This setup specifically requires the LV2 as it's part of the LV2 spec. And for that you need to run mod-ui as this is what uses the modgui.

**Reply by protectorateindustrial** [reply]
Could I run Cardinal in modep as plugin? I know there's an LV2 for offical Mod Devices, but could I compile a version that works in say piStomp?

**Reply by dromer** [reply]
Why not?

**Reply by protectorateindustrial** [reply]
Would I be able to control Cardinal remotely somehow? I mean the official Mod Devices version communicates through usb networking (with a static IP) to the Cardinal instance which is used to patch modules and stuff. If I were to use a rpi4 without the usb networking would the Cardinal instance used  as an editor be able to connect to the mod lv2?

**Reply by falkTX** [reply]
not with the web gui, but with a standalone remote tool yes.
I have been working on one for 23.01 release, though the "mini" variant takes priority (which is a special LV2 plugin version that allows remote control). Features made for this "mini" version will trickle down to the generic remote version.

**Reply by protectorateindustrial** [reply]
Ok, cool. Thats exactly the type of thing I was looking for. Standalone remote is just as good as a webgui :D

**Reply by protectorateindustrial** [reply]
I got the mod plugin working in my pistomp install. Would be cool to be able to change the ip where Cardinal on my desktop tries to connect. Just saying ;)
